components/tournament/tournament.lua

```diff
@@ -0,0 +1,180 @@
+		region = record.region,
+		status = record.status,
+		featured = recordIsFeatured(record),
+		icon = record.icon,
+		iconDark = record.icondark,
+		abbreviation = record.abbreviation,
+		series = record.series,
+	}
+
+	tournament.phase = Tournaments.calculatePhase(tournament)
+
+	return tournament
+end
+
+---@param tournament StandardTournament
+---@return TournamentPhase
+function Tournaments.calculatePhase(tournament)
+	if tournament.status == 'finished' then
+		return TOURNAMENT_PHASE.FINISHED
+	end
+	if not tournament.startDate then
+		return TOURNAMENT_PHASE.UPCOMING
+	end
+	if DateExt.getCurrentTimestamp() < tournament.startDate.timestamp then
+		return TOURNAMENT_PHASE.UPCOMING
+	end
+	if not tournament.endDate.timestamp then
+		return TOURNAMENT_PHASE.ONGOING
+	end
+	if DateExt.getCurrentTimestamp() < (tournament.endDate.timestamp + 24 * 60 * 60) then
+		return TOURNAMENT_PHASE.ONGOING
+	end
+	return TOURNAMENT_PHASE.FINISHED
+end
+
+--- This function parses fuzzy dates into a structured format.
+---@param dateRecord string # date in the format of `YYYY-MM-DD`, with `-MM-DD` optional.
+---@return {year: integer, month: integer?, day: integer?, timestamp: integer?}?
+function Tournaments.parseDateRecord(dateRecord)
+	if not dateRecord then
+		return nil
+	end
+	if dateRecord == DateExt.defaultDate then
+		return nil
+	end
+	local year, month, day = dateRecord:match('^(%d%d%d%d)%-?(%d?%d?)%-?(%d?%d?)')
+	year, month, day = tonumber(year), tonumber(month), tonumber(day)
+
+	if not year then
+		return
+	end
+
+	return {year = year, month = month, day = day, timestamp = DateExt.readTimestampOrNil(dateRecord)}
+end
+
+--- This function returns a function that can be used to determine if a tournament should be featured.
+---@return fun(record: tournament): boolean
+function Tournaments.makeFeaturedFunction()
+	local curatedData = Lua.requireIfExists('Module:TournamentsList/CuratedData', {loadData = true})
+	if not curatedData then
+		return function() return false end
+	end
+
+	---@param record tournament
+	---@return boolean
+	local function recordIsCurated(record)
+		if Table.includes(curatedData.exclude, record.pagename) then
+			return false
+		end
+		if Table.includes(curatedData.include, record.pagename) then
+			return true
+		end
+		if Table.includes({1, 2}, record.liquipediaTier) then
+			return true
+		end
+
+		if Logic.isEmpty(record.liquipediaTierType) then
+			return false
+		end
+
+		local function parentData(page, maxDepth)
+			if maxDepth == 0 then
+				return nil
+			end
+			local parentPage = mw.title.new(record.pagename).basePageTitle
+			return mw.ext.LiquipediaDB.lpdb('tournament', {
+				conditions = '[[pagename::' .. parentPage .. ']]',
+				limit = 1,
+			})[1] or parentData(parentPage, maxDepth - 1)
+		end
+		local parentTournament = parentData(record.pagename, 2)
+
+		if not parentTournament then
+			return false
+		end
+		return recordIsCurated(parentTournament)
+	end
+
+	return recordIsCurated
+end
```

Comment:
fwiw currently several wikis use different criteria for featuring stuff, e.g. a set publishertier

Reply:
It's the current "curated list", I renamed it featured in the code. Which is unrelated to highlighting